benchbuild/source/http.py

```diff
@@ -0,0 +1,69 @@
+"""
+Declare a http source.
+"""
+from typing import List, Mapping, Union
+
+import attr
+from plumbum import local
+
+from benchbuild import variants
+from benchbuild.source import base
+from benchbuild.utils.cmd import cp, tar, wget
+
+
+@attr.s
+class HTTP(base.BaseSource):
+    """
+    Fetch the downloadable source via http.
+    """
+
+    @property
+    def default(self) -> variants.Variant:
+        return self.versions()[0]
+
+    def version(self, target_dir: str, version: str) -> str:
+        prefix = base.target_prefix()
+        remotes = normalize_remotes(self.remote)
+
+        url = remotes[version]
+        target_name = versioned_target_name(self.local, version)
+        cache_path = local.path(prefix) / target_name
+        download_single_version(url, cache_path)
+
+        # FIXME: Belongs to environment code.
+
+        target_path = local.path(target_dir) / self.local
+        cp('-ar', cache_path, target_path)
+        return target_path
+
+    def versions(self) -> List[variants.Variant]:
+        remotes = normalize_remotes(self.remote)
+        return [variants.Variant(version=rev, owner=self) for rev in remotes]
+
+
+def normalize_remotes(
+        remote: Union[str, Mapping[str, str]]) -> Mapping[str, str]:
+    if not isinstance(remote, Mapping):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Second argument of isinstance is not a type](https://app.codacy.com/manual/polyjit/benchbuild_2/pullRequest?prid=5694346)